### DISCUSSION TITLE
Where can I download the style file ?

### DISCUSSION THREAD

**Discussion by moonsh** [cifkao/groove2groove#14] [Q&A] [answered]
I downloaded the data set you shared via this website : [https://zenodo.org/record/3958000#.YNG5YXUzaV4]

I found this midi file like "maj500.BRITROK8_b.midi"

I guess maj500 is content and BRITROK8 is a style file. 
And this midi file is an output of style embedding. Is that correct?

But not sure why there is a substyle a or b.
Could you explain this?
and where can I download the style file BRITROK8?

Thank you

**Comment by cifkao** [comment, marked as answer]
`maj500` would be the name of the chord chart, you can find it under `charts/test/abc/maj500.abc`. `BRITROK8` is the name of a Band-in-a-Box (BIAB) style – it comes with BIAB and you need BIAB/RealBand to use it. And `maj500.BRITROK8_b.mid` is the accompaniment produced by BIAB when given this chord chart as input and with the BRITROK8 style selected.

Band-in-a-Box styles moreover contain two substyles, which you can switch between in your chord chart. For example, you could use the A substyle for the first part of the song and the B substyle for the second part of the song. Here, we just use a single substyle for the whole chord chart, that's why we append the name of the substyle to the filename.

**Reply by cifkao** [reply]
To answer your question about the style file: The only situation where you would need this is if you wanted to generate more accompaniments in this style using BIAB. Then you would obviously need to buy BIAB and the style file should come with it.

**Comment by moonsh** [comment]
I see. Without buying BIAB, I can't get style files ?

Get Outlook for Android<https://aka.ms/AAb9ysg>

________________________________

**Reply by cifkao** [reply]
No, you can't. And without BIAB you also cannot use them, they only work with BIAB.

**Reply by cifkao** [reply]
Why would you want the style file anyway if you don't have BIAB?

**Reply by moonsh** [reply]
I just want to check how the input is changing after style transfer. To confirm style transfer works well, I want to hear style and input files.

**Reply by cifkao** [reply]
I think you are misunderstanding what a style file is. In BIAB, a style file (*.STY) is a file that contains a set of rules/patterns, created by the authors of BIAB, used to generate an accompaniment. It is not a MIDI file that you could play.

If you want to hear the inputs and targets used for training/evaluating Groove2Groove, just look at the MIDI files that I provide. For example, if `maj500.BRITROK8_b.mid` is a content input, then `maj612.SantaMon_a.mid` could be a style input and `maj500.SantaMon_a.mid` would be the target. I suggest you take a look at sections III.B and III.C in the paper, as well as Fig. 1 and Fig. 3, to understand how this works.

**Reply by moonsh** [reply]
Yes. Thank you for the detailed explanation. I will look as you said.
